packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -261,434 +263,353 @@ describe("BnsConnection", () => {
+      expect(tx).toEqual(sendTx);
+      // make sure we have a txid
+      expect(mine.txid).toBeDefined();
+      expect(mine.txid.length).toBeGreaterThan(0);
 
-    connection.disconnect();
-  });
+      connection.disconnect();
+    });
 
-  // Activate when https://github.com/tendermint/tendermint/issues/2759 is fixed
-  xit("can search for transactions (min/max height)", async () => {
-    pendingWithoutBnsd();
-    const connection = await BnsConnection.establish(bnsdTendermintUrl);
-    const chainId = await connection.chainId();
-    const initialHeight = await connection.height();
+    it("can post transaction and watch confirmations", done => {
+      pendingWithoutBnsd();
 
-    const { profile, mainWalletId, faucet } = await userProfileWithFaucet();
-    const faucetAddress = keyToAddress(faucet.pubkey);
-    const rcpt = await profile.createIdentity(mainWalletId, HdPaths.simpleAddress(68));
-    const rcptAddress = keyToAddress(rcpt.pubkey);
+      (async () => {
+        const connection = await BnsConnection.establish(bnsdTendermintUrl);
+        const chainId = await connection.chainId();
+
+        const { profile, mainWalletId, faucet } = await userProfileWithFaucet();
+        const faucetAddr = keyToAddress(faucet.pubkey);
+        const rcpt = await profile.createIdentity(mainWalletId, HdPaths.simpleAddress(2));
+        const rcptAddr = keyToAddress(rcpt.pubkey);
+
+        // construct a sendtx, this is normally used in the MultiChainSigner api
+        const sendTx: SendTx = {
+          kind: TransactionKind.Send,
+          chainId,
+          signer: faucet.pubkey,
+          recipient: rcptAddr,
+          memo: "My first payment",
+          amount: {
+            whole: 500,
+            fractional: 75000,
+            tokenTicker: cash,
+          },
+        };
+        const nonce = await getNonce(connection, faucetAddr);
+        const signed = await profile.signTransaction(mainWalletId, faucet, sendTx, bnsCodec, nonce);
+        const txBytes = bnsCodec.bytesToPost(signed);
+        const heightBeforeTransaction = await connection.height();
+        const result = await connection.postTx(txBytes);
+        expect(result.blockInfo!.value).toEqual({
+          state: BcpTransactionState.InBlock,
+          height: heightBeforeTransaction + 1,
+          confirmations: 1,
+        });
 
-    // construct a sendtx, this is normally used in the MultiChainSigner api
-    const memo = `Payment ${Math.random()}`;
-    const sendTx: SendTx = {
-      kind: TransactionKind.Send,
-      chainId: chainId,
-      signer: faucet.pubkey,
-      recipient: rcptAddress,
-      memo: memo,
-      amount: {
-        whole: 1,
-        fractional: 1,
-        tokenTicker: cash,
-      },
-    };
+        const events = new Array<BcpBlockInfo>();
+        const subscription = result.blockInfo!.updates.subscribe({
+          next: info => {
+            events.push(info);
+
+            if (events.length === 3) {
```

Comment:
Good test.

But everytime I see such code, I wonder if we could write a nice utility to help with streams in Promise land (when we want to await on some condition). Maybe some pieces from https://github.com/iov-one/iov-core/blob/master/packages/iov-stream/src/reducer.ts

Like:

```ts
const collector= asArray(result.blockInfo!.updates.take(3))
await collector.finished();
const events = collector.value();
expect(events.length).toEqual(3);
expect(events[0]).toEqual(...)
```

I think the take(3) will also handle unsubscribing